examples/customresources/demos/remoteserver/server_test.go

```diff
@@ -83,8 +83,14 @@ func TestGizmo(t *testing.T) {
 		test.That(t, r2.Close(context.Background()), test.ShouldBeNil)
 	}()
 	test.That(t, err, test.ShouldBeNil)
-	res, err := r2.ResourceByName(gizmoapi.Named("gizmo1"))
-	test.That(t, err, test.ShouldBeNil)
+
+	// remotes can take a few seconds to show up, so we wait for the resource
+	var res interface{}
+	testutils.WaitForAssertionWithSleep(t, time.Second, 120, func(tb testing.TB){
```

Comment:
Could it really take 2 minutes for the remote to show up? If it's supposed to only take a few seconds, how would you feel about checking every 0.1 seconds, so we give up after 12 seconds? That way, we react faster and give up faster. 

(You know more about this than I do. If we might need more than a minute to start up, this is fine as-is.)